### PR TITLE
2nd try: AlphaZero prep

### PR DESCRIPTION
**Pull request by alejandromarcu**
Basic skeleton for AlphaZero, and some framework changes to allow to have the 2 instances related. 
E.g. you would train it with: `/train.py -p alphazero:...`, and internally it would create 2 instances of the AlphaZero agent, one in training mode and one in play mode, and they will share the NN, temperature and probably other parameters as we realize we need.

### CODE REVIEW

File: deep_quoridor/src/agents/alphazero.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        super().__init__(**kwargs)
+        self.board_size = board_size
+        self.max_walls = max_walls
+        self.params = params
+        self.episode_count = 0
+        self.training_mode = training_mode
+
+        # If a training instance is passed, this instance is playing to train the other, sharing the NN and temperature
+        if training_instance:
+            self.nn = training_instance.nn
```
Comment (jonbinney) [inline deep_quoridor/src/agents/alphazero.py:47]: I like this concept of a "training instance" object with the context needed

**Review by jonbinney — APPROVED** [review]
This looks good to me. We can always modify it as needed.
